### ISSUE TITLE
Add Atom config to suppress a list of tslint errors from diagnostics

### ISSUE THREAD

**Issue by ryanberckmans** [AtomLinter/linter-tslint#260]
I use tslint with "run fix on save"; the tslint rule [quotemark](https://palantir.github.io/tslint/rules/quotemark/) is useful, but having it show up as a linter error in Atom/Nuclide diagnostics is prohibitively distracting for me. I'd rather have it silently fixed. One way to achieve this is to have an editor-defined blacklist of rules that can still be fixed, but don't show up in UI diagnostics.

**Comment by ryanberckmans** [comment]
This would bring tslint development closer to golang's glorious distraction-free development with `gofmt`. Another rule that would be useful to blacklist from UI diagnostics is [whitespace](https://palantir.github.io/tslint/rules/whitespace/).

**Comment by ryanberckmans** [comment]
It'd also be useful to have a checkbox to `Enable blacklist if and only if "Fix errors on save" is active`.

A blacklist of reasonably spammy rules could ship with `linter-tslint`, and this ^ option would be checked by default, such that new users would get a convenient blacklist, but by default the blacklist is only active if the user has opted into "Fix errors on save".

**Comment by Arcanemagus** [comment]
Over in `linter-eslint` we [send the current rules](https://github.com/AtomLinter/linter-eslint/blob/0de18656e42c1719f46f5db64c1171c3f5bc2101/src/worker.js#L23) whenever they have updated since the last time we sent them back to the main process. Back in the main process there is a [routine](https://github.com/AtomLinter/linter-eslint/blob/0de18656e42c1719f46f5db64c1171c3f5bc2101/src/rules.js#L30-L37) that returns the list of rules currently known to be fixable. If the user has enabled the option to ignore fixable rules, we use that list of fixable rules to just [tell ESLint to ignore them](https://github.com/AtomLinter/linter-eslint/blob/0de18656e42c1719f46f5db64c1171c3f5bc2101/src/main.js#L241-L246) for the lint jobs we run from the editor.

This process _does_ require at least one lint job to run before fixable rules are ignored, but after that it is *dynamic* what rules are ignored. Is something like this possible with `tslint`?

There is also an option that allows users to ignore any rule they specified, similar to your blacklist above. That should be blank though, this definitely shouldn't ship with a "default list".

**Comment by ryanberckmans** [comment]
> If the user has enabled the option to ignore fixable rules, we use that list of fixable rules to just tell ESLint to ignore them for the lint jobs we run from the editor.

@Arcanemagus nice! I like the idea of a config checkbox `Ignore (Don't show UI warnings for) Fixable Rules` as an alternative to my entire proposal. Not sure I'd ever want to ignore a non-fixable rule.

**Comment by Arcanemagus** [comment]
Functionality like that depends on being able to determine the list of fixable rules though, which I don't know if `tslint` provides that capability.

**Comment by Arcanemagus** [comment]
Note also that in `linter-eslint` _all_ "ignore these rules" options are ignored for lints on a buffer that has been saved, so users can still see messages that will fail CI checks if they have stopped editing the file.

**Comment by ryanberckmans** [comment]
@Arcanemagus makes sense. I'm open to any solution that suppresses fixable/blacklist of annoying linter errors, to make Atom typescript development less distracting like golang with auto `gofmt`.

If you're interested to propose a specific design for a solution, I'm happy to fund a gitcoin bounty for somebody to execute that design.
